src/Tasks/Microsoft.NET.Build.Tasks/CheckForDuplicateItems.cs

```diff
@@ -27,11 +27,18 @@ public class CheckForDuplicateItems : TaskBase
         [Required]
         public string MoreInformationLink { get; set; }
 
+        [Output]
+        public ITaskItem [] DeduplicatedItems { get; set; }
+
         protected override void ExecuteCore()
         {
+            DeduplicatedItems = Array.Empty<ITaskItem>();
+
             if (DefaultItemsEnabled && DefaultItemsOfThisTypeEnabled)
             {
-                var duplicateItems = Items.GroupBy(i => i.ItemSpec).Where(g => g.Count() > 1).ToList();
+                var itemGroups = Items.GroupBy(i => i.ItemSpec);
```

Comment:
>i => i.ItemSpec [](start = 47, length = 15)

For non-windows platforms, I think we need to handle "duplicate" items with different casing (e.g. Class1.cs and class1.cs) by making this `Items.GroupBy(i => i.ItemSpec, StringComparer.OrdinalIgnoreCase)`